test/lib/build/ProjectBuilder.js

```diff
@@ -471,6 +471,214 @@ test.serial("_buildProject: Requested project is not added to readers", async (t
+	mockProject.getReader = getReaderStub;
+
+	const getTagStub = sinon.stub().returns(false).onFirstCall().returns(true);
+	const projectBuildContextMock = {
+		getProject: () => mockProject,
+		getTaskUtil: () => {
+			return {
+				isRootProject: () => true,
+				getTag: getTagStub,
+				STANDARD_TAGS: {
+					OmitFromBuildResult: "OmitFromBuildResultTag"
+				}
+			};
+		}
+	};
+	const writerMock = {
+		write: sinon.stub().resolves()
+	};
+
+	await builder._writeResults(projectBuildContextMock, writerMock);
+
+	t.is(getReaderStub.callCount, 1, "One reader requested");
+	t.deepEqual(getReaderStub.getCall(0).args[0], {
+		style: "buildtime"
+	}, "Reader requested expected style");
+
+	t.is(byGlobStub.callCount, 1, "One byGlob call");
+	t.is(byGlobStub.getCall(0).args[0], "/**/*", "byGlob called with expected pattern");
+
+	t.is(createBuildManifestStub.callCount, 1, "createBuildManifest got called once");
+	t.is(createBuildManifestStub.getCall(0).args[0], mockProject,
+		"createBuildManifest got called with correct project");
+	t.deepEqual(createBuildManifestStub.getCall(0).args[1], {
+		createBuildManifest: true,
+		cssVariables: false,
+		excludedTasks: [],
+		includedTasks: [],
+		jsdoc: false,
+		selfContained: false,
+	}, "createBuildManifest got called with correct build configuration");
+
+	t.is(createResourceStub.callCount, 1, "One resource has been created");
+	t.deepEqual(createResourceStub.getCall(0).args[0], {
+		path: "/.ui5/build-manifest.json",
+		string: `{
+	"build": "manifest"
+}`
+	}, "Build manifest resource has been created with correct arguments");
+
+	t.is(getTagStub.callCount, 3, "TaskUtil#getTag got called three times");
+	t.is(getTagStub.getCall(0).args[0], dummyResources[0], "TaskUtil#getTag called with first resource");
+	t.is(getTagStub.getCall(0).args[1], "OmitFromBuildResultTag", "TaskUtil#getTag called with correct tag value");
+	t.is(getTagStub.getCall(1).args[0], dummyResources[1], "TaskUtil#getTag called with second resource");
+	t.is(getTagStub.getCall(1).args[1], "OmitFromBuildResultTag", "TaskUtil#getTag called with correct tag value");
+	t.is(getTagStub.getCall(2).args[0], dummyResources[2], "TaskUtil#getTag called with third resource");
+	t.is(getTagStub.getCall(2).args[1], "OmitFromBuildResultTag", "TaskUtil#getTag called with correct tag value");
+
+	t.is(writerMock.write.callCount, 3, "Write got called three times");
+	t.is(writerMock.write.getCall(0).args[0], "build manifest resource", "Write got called with build manifest");
+	t.is(writerMock.write.getCall(1).args[0], dummyResources[1], "Write got called with second resource");
+	t.is(writerMock.write.getCall(2).args[0], dummyResources[2], "Write got called with third resource");
+});
+
+test("_writeResults: Do not create build manifest for non-root project", async (t) => {
+	t.context.getRootTypeStub = sinon.stub().returns("library");
+	const {graph} = t.context;
+
+	const ProjectBuilder = mock.reRequire("../../../lib/build/ProjectBuilder");
```

Comment:
Is this re-require necessary? If so, I'd rather expect this to be a `serial` test

Reply:
You're right, thanks! 👍